### PR TITLE
Use sjlf4j-api instead of impl in SDK

### LINKED ISSUE #271: maven artifacts for ASK SDK for Java should not depend on an SLF4J binding, but only the API
## I'm submitting a...

<pre><code>[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

Libraries should not depend on an SLF4J binding, as that imposes a specific choice of logging implementation on clients of the library.

Please read this section of the SLF4J docs: http://www.slf4j.org/codes.html#multiple_bindings

Specifically note this paragraph at the end of that section:
> Embedded components such as libraries or frameworks should not declare a dependency on any SLF4J binding but only depend on slf4j-api. When a library declares a compile-time dependency on a SLF4J binding, it imposes that binding on the end-user, thus negating SLF4J's purpose. When you come across an embedded component declaring a compile-time dependency on any SLF4J binding, please take the time to contact the authors of said component/library and kindly ask them to mend their ways.

## Expected Behavior

The pom.xml files for the ASK SDK artifacts should depend on slf4j-api, and not log4j-slf4j-impl.

## Current Behavior

Currently, these pom.xml files include a dependency on log4j-slf4j-impl:
https://repo1.maven.org/maven2/com/amazon/alexa/ask-sdk-runtime/2.35.0/ask-sdk-runtime-2.35.0.pom
https://repo1.maven.org/maven2/com/amazon/alexa/ask-sdk-model-runtime/1.0.4/ask-sdk-model-runtime-1.0.4.pom

## Possible Solution
Replace log4j-slf4j-impl dependency with slf4j-api dependency, and let the clients of your library supply the implementation of SLF4J that matches their underlying logging system (log4j v1, log4j v2, Logback, etc.)

## Context

We use log4j v1, and provide the log4j v1 SLF4J implementation (slf4j-log4j12) in our classpath, but because of the ASK SDK for Java's dependency declarations, maven transitively includes log4j-slf4j-impl, resulting in SLF4J having two competing implementations on the classpath, only one of which is valid. This leads to the "multiple bindings" error at startup (described in the link above), and then any libraries (including ASK) which log via SLF4J do not get their log messages routed to log4j (instead they fall back to stderr).

## Your Environment

* ASK Java SDK version used: 2.30.0
* Operating System and version: Ubuntu Linux 18.04

## Java Info

* Java version used for development: openjdk version "1.8.0_265"

### PR DESCRIPTION
**Pull request by kkocel**
Use log4j-slf4j-impl in samples in tests.
Logging instructtions in the Readme.md

Fixes #271

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License
- [x] By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

[issues]: https://github.com/alexa/alexa-skills-kit-sdk-for-java/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: ask-sdk-local-debug/tst/com/amazon/ask/localdebug/client/WebSocketClientImplTest.java [hunk 1]
```diff
@@ -46,6 +47,7 @@
 @PrepareForTest({
         RequestResponseUtils.class
 })
+@PowerMockIgnore({"javax.management.*", "javax.script.*"})
```
Comment (Shreyas-vgr) [inline ask-sdk-local-debug/tst/com/amazon/ask/localdebug/client/WebSocketClientImplTest.java]: What is this Ignore for ?

Also, I see you applied it across multiple packages
Maybe use a configuration file instead for eg: https://github.com/powermock/powermock-examples-maven/blob/master/global-ignore/src/test/resources/org/powermock/extensions/configuration.properties

Comment (kkocel) [reply ask-sdk-local-debug/tst/com/amazon/ask/localdebug/client/WebSocketClientImplTest.java]: without those ignores there would be power mock errors reported. Like:
```
ScriptEngineManager providers.next(): javax.script.ScriptEngineFactory: Provider jdk.nashorn.api.scripting.NashornScriptEngineFactory not a subtype
ERROR StatusLogger Could not reconfigure JMX
 java.lang.LinkageError: loader constraint violation: loader (instance of org/powermock/core/classloader/javassist/JavassistMockClassLoader) previously initiated loading for a different type with name "javax/management/MBeanServer"
```
I added them to the global config file as you suggested.

**Comment by kkocel** [comment]
@Shreyas-vgr so it's good to merge?

**Comment by SCKelemen** [comment]
Maybe this should be prioritized:
https://www.lunasec.io/docs/blog/log4j-zero-day/

**Comment by kkocel** [comment]
@SCKelemen yes, I bumped dependencies to fix this 0 day, @Shreyas-vgr another reason for the merge :)

Fortunately, my PR removes log4j impl from the production code so log4j won't be added into the user codebase.

**Comment by htmldoug** [comment]
@ShenChen93, sorry for the direct tag, but given the severity, can you please help get this merged and released?

**Review by Shreyas-vgr — APPROVED** [review]

**Review by ag-chirag — APPROVED** [review]
Looks good.

**Comment by Shreyas-vgr** [comment]
Merged the change to mainline, thanks @kkocel for the PR

**Pull request merged**